### PR TITLE
ASM-3917 Rename binary to avoid "Path too long"

### PR DESCRIPTION
**Pull request by logankimmel**

### CODE REVIEW

File: lib/puppet/provider/cmc_fw_update/racadm.rb [hunk 1]
```diff
@@ -151,6 +166,7 @@ def create
         retries += 1
       end
     end
+    remove_renamed_file
```
Comment (gavin-scott) [inline lib/puppet/provider/cmc_fw_update/racadm.rb]: This should really be in an ensure block to make sure it always runs.

File: lib/puppet/provider/cmc_fw_update/racadm.rb [hunk 2]
```diff
@@ -91,26 +91,41 @@ def transport
...
   end
 
+  def new_binary_name
+    random_string = (0...8).map { (65 + rand(26)).chr }.join
+    "CMC_#{random_string }.EXE"
+  end
+
+  def rename_binary
+    tftp_share = File.dirname(@fw['path'])
+    new_filename = new_binary_name
```
Comment (gavin-scott) [inline lib/puppet/provider/cmc_fw_update/racadm.rb]: Ideally we would check if new_filename already exists and generate a new name or append a _1, _2 or something if it did.

**Comment by gavin-scott** [comment]
Couple nits, but overall looks good. Were you able to test this end to end?

**Comment by logankimmel** [comment]
@gavin-scott I tested this end-to-end on 172.18.4.185  You can see the logs on there.  

My thinking was to keep new exe around if something fails, that way it's easier for troubleshooting.  However, I did add the debug message to the original binary location for that a similar reason.  

The odds of both the file not getting cleaned up and the random generated prefix being the same are super low.  If you really think we need to add a suffix like _1.. I can do that but I don't think it's necessary.

**Comment by gavin-scott** [comment]
Agree neither point should cause any practical problems.

**Pull request merged**